monitoring/sensu_check.py

```diff
@@ -0,0 +1,317 @@
+    required: false
+    default: 10
+  handle:
+    description: Whether the check should be handled or not
+    choices: [ 'yes', 'no' ]
+    required: false
+    default: yes
+  subdue_begin:
+    description: When to disable handling of check failures
+    required: false
+    default: null
+  subdue_end:
+    description: When to enable handling of check failures
+    required: false
+    default: null
+  dependencies:
+    description: >-
+      Other checks this check depends on, if dependencies fail,
+      handling of this check will be disabled
+    required: false
+    default: []
+  metric:
+    description: Whether the check is a metric
+    choices: [ 'yes', 'no' ]
+    required: false
+    default: no
+  standalone:
+    description: >-
+      Whether the check should be scheduled by the sensu client or server
+      This option obviates the need for specifying the I(subscribers) option
+    choices: [ 'yes', 'no' ]
+    required: false
+    default: no
+  publish:
+    description: >-
+      Whether the check should be scheduled at all.
+      You can still issue it via the sensu api
+    choices: [ 'yes', 'no' ]
+    required: false
+    default: yes
+  occurrences:
+    description: >-
+      Number of event occurrences before the handler should take action
+    required: false
+    default: 1
+  refresh:
+    description: >-
+      Number of seconds handlers should wait before taking second action
+    required: false
+    default: null
+  aggregate:
+    description: >-
+      Classifies the check as an aggregate check,
+      making it available via the aggregate API
+    choices: [ 'yes', 'no' ]
+    required: false
+    default: no
+  low_flap_threshold:
+    description: The low threshhold for flap detection
+    required: false
+    default: null
+  high_flap_threshold:
+    description: The low threshhold for flap detection
+    required: false
+    default: null
+version_added: 2.0
+requirements: [ ]
+author: Anders Ingemann
+'''
+
+EXAMPLES = '''
+# Fetch metrics about the CPU load every 60 seconds,
+# the sensu server has a handler called 'relay' which forwards stats to graphite
+- name: get cpu metrics
+  sensu_check: name=cpu_load
+               command=/etc/sensu/plugins/system/cpu-mpstat-metrics.rb
+               metric=yes handlers=relay subscribers=common interval=60
+
+# Check whether nginx is running
+- name: check nginx process
+  sensu_check: name=nginx_running
+               command='/etc/sensu/plugins/processes/check-procs.rb -f /var/run/nginx.pid'
+               handlers=default subscribers=nginx interval=60
+
+# Stop monitoring the disk capacity.
+# Note that the check will still show up in the sensu dashboard,
+# to remove it completely you need to issue a DELETE request to the sensu api.
+- name: check disk
+  sensu_check: name=check_disk_capacity
+'''
+
+def sensu_check(module, path, name, state='present', backup=False):
+    changed = False
+    reasons = []
+
+    import json
```

Comment:
this may run on older hosts that do not have json but simplejson, use a exception handling to default to the latter, check other modules as an example